qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -136,94 +138,132 @@ def __init__(self, width: ParameterValueType):
-                to scan, in units of dt. The total pulse duration including Gaussian rising and
-                falling edges is implicitly computed with experiment parameters ``sigma`` and
-                ``risefall``.
+            flat_top_widths: Deprecated. The total duration of the square part of
+                cross resonance pulse(s) to scan, in units of dt.
+                The total pulse duration including Gaussian rising and falling edges is
+                implicitly computed with experiment parameters ``sigma`` and ``risefall``.
             backend: Optional, the backend to run the experiment on.
-            cr_gate: Optional, circuit gate instruction of cross resonance pulse.
+            cr_gate: Optional, circuit gate class representing the cross resonance pulse.
+                Providing this object allows us to run this experiment with circuit simulator,
+                and this object might be used for testing, development of analysis protocol,
+                and educational purpose without needing to wait for hardware queueing.
+                Note that this instance must provide matrix representation, such as
+                unitary gate or Hamiltonian gate, and the class is expected to be instantiated
+                with a single parameter ``width`` in units of sec.
+            durations: Optional. The total duration of cross resonance pulse(s) including
+                rising and falling edges. The minimum number should be larger than the
+                total lengths of these ramps. If not provided, then ``num_durations`` evenly
+                spaced durations between ``min_durations`` and ``max_durations`` are
+                automatically generated from these experiment options. The default numbers
+                are chosen to have a good sensitivity for the Hamiltonian coefficient
+                of interest at the rate around 1 MHz.
+                This argument should be provided in units of sec.
             kwargs: Pulse parameters. See :meth:`experiment_options` for details.
 
         Raises:
             QiskitError: When ``qubits`` length is not 2.
         """
-        # backend parameters required to run this experiment
-        # random values are populated here but these are immediately updated after backend is set
-        # this is to keep capability of generating circuits just for checking
-        self._dt = 1
-        self._cr_channel = 0
-        self._granularity = 1
-        self._cr_gate = cr_gate
-
         if len(qubits) != 2:
             raise QiskitError(
                 "Length of qubits is not 2. Please provide index for control and target qubit."
             )
 
         super().__init__(qubits, analysis=CrossResonanceHamiltonianAnalysis(), backend=backend)
-        self.set_experiment_options(flat_top_widths=flat_top_widths, **kwargs)
+        self._backend_timing = None
```

Comment:
This should go above the call to `super` which is probably why the test is failing.

Reply:
Good catch. Partly fixed in https://github.com/Qiskit/qiskit-experiments/pull/933/commits/d4c9f4696e789b72072b2514332714977a821769.

I also needed to add an exception handling when backend is not set and circuit is called. This is always annoying and requires us to do some overengineering. https://github.com/Qiskit/qiskit-experiments/pull/933/commits/feb2101cb3040f807e1ce3aa03a3327441a6d61c